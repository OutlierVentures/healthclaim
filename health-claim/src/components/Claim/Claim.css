.record-text {
  font-size: 1em;
  align-self: flex-start;
  margin-top: 2em;
  margin-left: 3em;
}

.bar {
  background-color: black;
  width: 70em;
  height: 0.07em;
  margin-top: 4em;
  margin-right: 9em;
}

.make-claim {
  margin-left: 4em;
  margin-top: 1em;
}

.records-button {
  text-align: end;
  margin-top: 1em;
  margin-right: 19em;
}

.record-select {
  margin: auto;
}

.health-records {
  justify-self: center;
  align-self: center;
  display: flex;
  margin: 0 auto;
  width: 60em;
  margin-top: 3em;
}

.claim-button {
  margin-top: 7em;
  align-self: flex-end;
  background: linear-gradient(#0076ba, #56c1ff);
  margin-top: 3em;
  margin-left: 60em;
}
