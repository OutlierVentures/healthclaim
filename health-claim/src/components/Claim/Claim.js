import React, { Component } from "react";
import "./Claim.css";
import { Button } from "reactstrap";
import RecordCard from "../RecordCard/RecordCard";
import Header from "../Header/Header";
import "bootstrap/dist/css/bootstrap.css";

export default class Claim extends Component {
  handleClick() {}

  renderRecords(records) {
    return (
      <div className="health-records">
        {records.map(record => (
          <div>
            <RecordCard record={record} onClick={this.handleClick} />
          </div>
        ))}
      </div>
    );
  }

  render() {
    const healthRecord1 = {
      title: "Health record 1",
      description: "Issued by doctor Smith"
    };
    const healthRecord2 = {
      title: "Health record 2",
      description: "Issued by doctor Ben"
    };
    const healthRecord3 = {
      title: "Health record 3",
      description: "Issued by doctor Michael"
    };

    const records = [healthRecord1, healthRecord2, healthRecord3];

    return (
      <div>
        <div>
          <Header />
        </div>
        <div class="record-select">
          <h4 className="record-text">Choose one of your health records:</h4>
          {this.renderRecords(records)}
          <div className="records-button">
            <Button color="link">See older health records</Button>
          </div>
        </div>
        <div className="bar"> </div>
        <div className="make-claim">
          <h5> Make a claim </h5>
          <text> Choose the type of your claim: </text>
        </div>
        <Button className="claim-button">Post claim</Button>
      </div>
    );
  }
}
