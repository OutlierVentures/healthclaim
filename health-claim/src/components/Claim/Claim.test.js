import React from "react";
import renderer from "react-test-renderer";
import Claim from "./Claim.js";

describe("Claim", () => {
  it("should render correctly with mocked records", () => {
    const tree = renderer.create(<Claim />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
