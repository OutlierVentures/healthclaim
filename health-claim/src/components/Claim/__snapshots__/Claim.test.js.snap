// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claim should render correctly with mocked records 1`] = `
<div>
  <div>
    <div>
      <nav
        className="navbar navbar navbar-expand-md navbar-light"
      >
        <ul
          className="nav"
        >
          <li
            className="nav-item"
          >
             
            <img
              alt="logo"
              src="./icon.png"
            />
          </li>
          <li
            className="header nav-item"
          >
            HEALTH CLAIMS
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    class="record-select"
  >
    <h4
      className="record-text"
    >
      Choose one of your health records:
    </h4>
    <div
      className="health-records"
    >
      <div>
        <div
          className="card card"
          onClick={[Function]}
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Health record 1
            </h5>
            <p
              className="card-text"
            >
              Issued by doctor Smith
            </p>
            <button
              aria-label={null}
              className="btn btn-link"
              onClick={[Function]}
            >
              Open the record
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          className="card card"
          onClick={[Function]}
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Health record 2
            </h5>
            <p
              className="card-text"
            >
              Issued by doctor Ben
            </p>
            <button
              aria-label={null}
              className="btn btn-link"
              onClick={[Function]}
            >
              Open the record
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          className="card card"
          onClick={[Function]}
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Health record 3
            </h5>
            <p
              className="card-text"
            >
              Issued by doctor Michael
            </p>
            <button
              aria-label={null}
              className="btn btn-link"
              onClick={[Function]}
            >
              Open the record
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="records-button"
    >
      <button
        aria-label={null}
        className="btn btn-link"
        onClick={[Function]}
      >
        See older health records
      </button>
    </div>
  </div>
  <div
    className="bar"
  >
     
  </div>
  <div
    className="make-claim"
  >
    <h5>
       Make a claim 
    </h5>
    <text>
       Choose the type of your claim: 
    </text>
  </div>
  <button
    aria-label={null}
    className="claim-button btn btn-secondary"
    onClick={[Function]}
  >
    Post claim
  </button>
</div>
`;
